File: JavaScript/10.js
function generateWeatherData(days){
    var data = [];
    for(var d=0; d<days; d++){
        var record = {};
        record.day = d+1;
        record.temp = Math.random()*45 - 10;
        record.humidity = Math.random()*80 + 20;
        record.wind = Math.random()*20;
        data.push(record);
    }
    return data;
    }
    function averageTemperature(data){
    var total = 0;
    for(var i=0; i<data.length; i++){
        total += data[i].temp;
    }
    return data.length > 0 ? total/data.length : 0;
    }
    function maxWind(data){
    var max = 0;
    for(var i=0; i<data.length; i++){
        if(data[i].wind > max){
        max = data[i].wind;
        }
    }
    return max;
    }
    function filterHotDays(data, threshold){
    var hot = [];
    for(var i=0; i<data.length; i++){
        if(data[i].temp > threshold){
        hot.push(data[i]);
        }
    }
    return hot;
    }
    function printWeatherData(data){
    for(var i=0;i<data.length;i++){
        console.log("Day "+data[i].day+", Temp: "+data[i].temp+", Humidity: "+data[i].humidity+", Wind: "+data[i].wind);
    }
    }
    var weather = generateWeatherData(15);
    console.log("Weather Data:");
    printWeatherData(weather);
    var avgTemp = averageTemperature(weather);
    console.log("Average Temperature: " + avgTemp);
    var maxWindSpeed = maxWind(weather);
    console.log("Maximum Wind Speed: " + maxWindSpeed);
    var hotDays = filterHotDays(weather, 25);
    console.log("Hot Days:");
    printWeatherData(hotDays);
    for(var i=0;i<3;i++){
    for(var j=0;j<2;j++){
        console.log("Nested loop "+i+","+j);
    }
    }
    var totalRecords = weather.length;
    console.log("Total records: " + totalRecords);
    var sumTemp = 0;
    for(var i=0;i<weather.length;i++){
    sumTemp += weather[i].temp;
    }
    var avgTemp2 = totalRecords > 0 ? sumTemp/totalRecords : 0;
    console.log("Calculated Average Temperature: " + avgTemp2);
    for(var a=0;a<2;a++){
    console.log("Extra loop "+a);
    }
    for(var b=0;b<2;b++){
    for(var c=0;c<2;c++){
        console.log("Double nested "+b+","+c);
    }
    }
    console.log("Additional analysis start");
    var tempList = [];
    for(var i=0;i<weather.length;i++){
    tempList.push(weather[i].temp);
    }
    console.log("Temperatures: " + tempList);
    var maxTemp = Math.max.apply(null, tempList);
    var minTemp = Math.min.apply(null, tempList);
    console.log("Max Temp: " + maxTemp);
    console.log("Min Temp: " + minTemp);
    var diff = maxTemp - minTemp;
    console.log("Temperature Difference: " + diff);
    for(var p=0;p<2;p++){
    console.log("Loop p "+p);
    }
    for(var q=0;q<2;q++){
    console.log("Loop q "+q);
    }
    for(var r=0;r<2;r++){
    for(var s=0;s<2;s++){
        console.log("Nested r,s "+r+","+s);
    }
    }
    console.log("Additional analysis complete");
    var finalSummary = {totalRecords: totalRecords, averageTemp: avgTemp2};
    console.log("Final summary: ", finalSummary);
    console.log("Simulation ends now");
    console.log("Goodbye");
    for(var i=0;i<2;i++){
    console.log("Final loop "+i);
    }
    console.log("End of Weather Data Analysis");

File: Python/10.py
import random
def generate_weather_data(days):
 data=[]
 for d in range(days):
  temp=random.uniform(-10,35)
  humidity=random.randint(20,100)
  wind=random.uniform(0,20)
  data.append({"day":d+1,"temp":temp,"humidity":humidity,"wind":wind})
 return data
def average_temperature(data):
 total=0
 for record in data:
  total+=record["temp"]
 return total/len(data) if data else 0
def max_wind(data):
 max_w=0
 for record in data:
  if record["wind"]>max_w:
   max_w=record["wind"]
 return max_w
def filter_hot_days(data,threshold):
 hot=[]
 for record in data:
  if record["temp"]>threshold:
   hot.append(record)
 return hot
def print_weather_data(data):
 for record in data:
  print("Day",record["day"],"Temp:",record["temp"],"Humidity:",record["humidity"],"Wind:",record["wind"])
weather=generate_weather_data(15)
print("Weather Data:")
print_weather_data(weather)
avg_temp=average_temperature(weather)
print("Average Temperature:",avg_temp)
max_wind_speed=max_wind(weather)
print("Maximum Wind Speed:",max_wind_speed)
hot_days=filter_hot_days(weather,25)
print("Hot Days:")
print_weather_data(hot_days)
for i in range(3):
 for j in range(2):
  print("Nested loop",i,j)
total_records=len(weather)
print("Total records:",total_records)
sum_temp=0
for record in weather:
 sum_temp+=record["temp"]
avg_temp2=sum_temp/total_records
print("Calculated Average Temperature:",avg_temp2)
for a in range(2):
 print("Extra loop",a)
for b in range(2):
 for c in range(2):
  print("Double nested",b,c)
print("Weather data analysis complete")
for m in range(3):
 for n in range(3):
  print("Additional nested",m,n)
print("Additional analysis start")
temp_list=[rec["temp"] for rec in weather]
print("Temperatures:",temp_list)
max_temp=max(temp_list)
min_temp=min(temp_list)
print("Max Temp:",max_temp)
print("Min Temp:",min_temp)
diff=max_temp-min_temp
print("Temperature Difference:",diff)
for p in range(2):
 print("Loop p",p)
for q in range(2):
 print("Loop q",q)
for r in range(2):
 for s in range(2):
  print("Nested r,s",r,s)
print("Additional analysis complete")
def final_summary():
 summary="Weather analysis complete. Total records: "+str(total_records)
 return summary
summ=final_summary()
print(summ)
print("Simulation ends now")
print("Goodbye")
for i in range(2):
 print("Final loop",i)
print("End of Weather Data Analysis")
for m in range(3):
 for n in range(3):
  print("Additional nested",m,n)
print("Additional analysis start")
temp_list=[rec["temp"] for rec in weather]
print("Temperatures:",temp_list)
max_temp=max(temp_list)
min_temp=min(temp_list)
print("Max Temp:",max_temp)
print("Min Temp:",min_temp)
diff=max_temp-min_temp
print("Temperature Difference:",diff)
for p in range(2):
 print("Loop p",p)
for q in range(2):
 print("Loop q",q)
for r in range(2):
 for s in range(2):
  print("Nested r,s",r,s)
print("Additional analysis complete")
def final_summary():
 summary="Weather analysis complete. Total records: "+str(total_records)
 return summary
summ=final_summary()
print(summ)
print("Simulation ends now")
print("Goodbye")
for i in range(2):
 print("Final loop",i)
print("End of Weather Data Analysis")
